ignite/pkg/cosmosgen/generate.go

```diff
@@ -2,40 +2,73 @@ package cosmosgen
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"io/fs"
 	"os"
 	"path/filepath"
+	"slices"
 	"strings"
 
 	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
 
 	"github.com/ignite/cli/ignite/pkg/cache"
+	"github.com/ignite/cli/ignite/pkg/cliui/colors"
+	"github.com/ignite/cli/ignite/pkg/cliui/icons"
 	"github.com/ignite/cli/ignite/pkg/cmdrunner"
 	"github.com/ignite/cli/ignite/pkg/cmdrunner/step"
 	"github.com/ignite/cli/ignite/pkg/cosmosanalysis/module"
 	"github.com/ignite/cli/ignite/pkg/cosmosbuf"
 	"github.com/ignite/cli/ignite/pkg/cosmosver"
+	"github.com/ignite/cli/ignite/pkg/events"
 	"github.com/ignite/cli/ignite/pkg/gomodule"
 	"github.com/ignite/cli/ignite/pkg/xfilepath"
+	"github.com/ignite/cli/ignite/pkg/xos"
 )
 
 const (
 	moduleCacheNamespace       = "generate.setup.module"
 	includeProtoCacheNamespace = "generator.includes.proto"
+	workFilename               = "buf.work.yaml"
 )
 
 var protocGlobalInclude = xfilepath.List(
 	xfilepath.JoinFromHome(xfilepath.Path("local/include")),
 	xfilepath.JoinFromHome(xfilepath.Path(".local/include")),
 )
 
-type ModulesInPath struct {
-	Path     string
-	Modules  []module.Module
-	Includes []string
+// protoIncludes contains proto include paths for a package.
+type protoIncludes struct {
+	// Paths is a list of proto include paths.
+	Paths []string
+
+	// BufPath is the path to the Buf config file when it exists.
+	BufPath string
+
+	// ProtoPath contains the path to the package's proto directory.
+	ProtoPath string
 }
 
-func (g *generator) setup() (err error) {
+// protoAnalysis contains proto module analysis data for a Go package dependency.
+type protoAnalysis struct {
+	// Path is the full path to the Go dependency
+	Path string
+
+	// Modules contains the proto modules analysis data.
+	// The list is empty when the Go package has no proto files.
+	Modules []module.Module
+
+	// Includes contain proto include paths.
+	// These paths should be used when generating code.
+	Includes protoIncludes
+}
+
+func newBufConfigError(path string, cause error) error {
+	return fmt.Errorf("invalid Buf config: %s: %w", path, cause)
```

Comment:
Can we create a `var` error that we can use in the `errors.Is` method? and we can wrap to add more info

Reply:
Changeset 10cfe6d